Preinitialize tracer for faster restore times

dettmoney opened the pull request:
Profiling criu restore time to first response using acmeair microservices applications shows time being spent in io/openliberty/opentracing/internal/OpentracingTracerManger.ensureTracer(String) during the first request to the application. Specifically in the `applicationTracers.computeIfAbsent(appName, TracerCreator.INSTANCE)` call. If we can move this call into the criu checkpoint phase when the application is started, we can reduce the restore time up to 20%.

joe-chacko committed:
Improve messaging singleton mechanism

Commit:
Preinitialize tracer for faster restore times

Preinitialize tracer for faster restore times

Preinitialize tracer for faster restore times

Preinitialize tracer for faster restore times

dettmoney commented:
#build

dettmoney commented:
I'm wondering if the fix for this one might be bring work into the checkpoint phase that we don't want there. Here's a stack that shows what's being called by the getTracer method:

```
[5/24/22, 22:12:36:913 EDT] 0000002d io.jaegertracing.internal.senders.SenderResolver             W Failed to get a sender from the sender factory.
java.lang.RuntimeException: TUDPTransport cannot connect:
        at io.jaegertracing.thrift.internal.reporters.protocols.ThriftUdpTransport.newThriftUdpClient(ThriftUdpTransport.java:50)
        at io.jaegertracing.thrift.internal.senders.UdpSender.<init>(UdpSender.java:57)
        at io.jaegertracing.thrift.internal.senders.ThriftSenderFactory.getSender(ThriftSenderFactory.java:36)
        at io.jaegertracing.internal.senders.SenderResolver.getSenderFromFactory(SenderResolver.java:110)
        at io.jaegertracing.internal.senders.SenderResolver.resolve(SenderResolver.java:88)
        at io.jaegertracing.Configuration$SenderConfiguration.getSender(Configuration.java:696)
        at io.jaegertracing.Configuration$ReporterConfiguration.getReporter(Configuration.java:593)
        at io.jaegertracing.Configuration$ReporterConfiguration.access$000(Configuration.java:553)
        at io.jaegertracing.Configuration.getTracerBuilder(Configuration.java:230)
        at io.jaegertracing.Configuration.getTracer(Configuration.java:253)
        at com.ibm.ws.microprofile.opentracing.jaeger.adapter.impl.ConfigurationImpl.getTracer(ConfigurationImpl.java:29)
        at com.ibm.ws.microprofile.opentracing.jaeger.JaegerTracerFactory.createJaegerTracer(JaegerTracerFactory.java:181)
        at io.openliberty.opentracing.internal.OpentracingTracerManager.createJaegerTracer(OpentracingTracerManager.java:78)
        at io.openliberty.opentracing.internal.OpentracingTracerManager.access$000(OpentracingTracerManager.java:35)
        at io.openliberty.opentracing.internal.OpentracingTracerManager$TracerCreator.apply(OpentracingTracerManager.java:87)
        at io.openliberty.opentracing.internal.OpentracingTracerManager$TracerCreator.apply(OpentracingTracerManager.java:81)
        at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
        at io.openliberty.opentracing.internal.OpentracingTracerManager.ensureTracer(OpentracingTracerManager.java:58)
        at io.openliberty.opentracing.internal.OpentracingTracerManager.getTracer(OpentracingTracerManager.java:164)
```

@tjwatson - if this call results in a client making a remote connection, then we would want it to happen during restore, correct?

tjwatson commented:
> @tjwatson - if this call results in a client making a remote connection, then we would want it to happen during restore, correct?

Correct, I think this is related to the issue #21237 I opened.  We cannot allow connections to persist across a checkpoint/restore.  It is pretty much guaranteed that will fail in a container scenario where restore is not done on the host that created the checkpoint.

jhanders34 committed:
Update transformer to version 0.5.0

tjwatson reviewed (commented):
Something is wrong with this PR.  I'm pretty sure you did not mean to include 843 commits with 4000+ file changes.

emilytee committed:
JakartaEE10Authorization

emilytee committed:
JakartaEE10Authorization

emilytee committed:
JakartaEE10Authorization

volosied committed:
create servercontainer10 / servercontainer21 packages
initial upgradeHttpToWebSocket behavior

Fix up upgrade servlet app
Do some renaming

Throw deployment exception

Fix up upgradeHttpToWebSocket  DeploymentException

dazey3 committed:
Remove com.ibm.ws.jpa.tests.spec20_fat FAT

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

dazey3 committed:
Create new com.ibm.ws.jpa.tests.spec20_fat FAT

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

joe-chacko committed:
Issue #22227: update Yoko for null field interoperability

volosied committed:
create and build full URI for #20585

volosied committed:
Various corrections & create getRequestURI test

gjwatts committed:
Java 19 FAT test

emilytee committed:
JakarataEE10Authorization

arunavemulapalli committed:
oidc identity store update (1)

pmd1nh committed:
FAT for Cookie setAttribute

Azquelt committed:
Update mpOpenAPI 3.1 for config validation

* Update the config validation autofeatures and bundles to work with
  mpOpenAPI-3.1.
* Update the config validation FAT tests to run against EE10

pgunapal committed:
Merge pull request #22260 from pgunapal/21868-repeatOnlyBasicRTTests

Skip Repeat for certain redundant RequestTiming FATs

tjwatson committed:
Merge pull request #22316 from tjwatson/setCheckpointLogLevel

Allow CRIU_LOG_LEVEL to control criu log level for checkpoint

neilgsyoung committed:
Tidy up stale lease in tx failover tests

pnicolucci committed:
Initial code checkin

c00crane committed:
disable jakartasec.3.0 while product code is developed

jjiwooLim committed:
fix repo connection failure

ayoho committed:
Merge pull request #22273 from ayoho/jakarta-21165-2

Update Jakarta Security 3.0 cookie storage behavior to be in line with spec

ayoho committed:
Merge pull request #22263 from ayoho/gsonRemoval

Remove project and remaining references to old GSON library

sebratton committed:
Add to bnd buildpath to address an eclipse compile issue.

ayoho committed:
Issue 22300: Fix issue with empty strings returned from provider metadata

Resolves #22300

njr-11 committed:
Merge pull request #22302 from njr-11/22301-JCABVTTest-is-not-expecting-repeat-on-EE9

JCABVTTest not expecting repeat on EE9

Commit:
Ignore invalid resources in checkAtomicity()

ayoho committed:
Issue 21157: Add SSL support for discovery

Updates the Jakarta OIDC flow to use the `SSLSocketFactory` obtained from the `SSLSupport` service when performing discovery.

For #21157

mswatosh committed:
Merge pull request #22119 from mswatosh/bb291835-timeout-increase

Failover tests timeout increase

njr-11 committed:
Issue #21063 updates to entity annotations for records

tjwatson committed:
Move configureEnvVariable to common method on FATSuite

Also made the method take a Map<String, String> to allow
more than one variable setting.

tjwatson committed:
Add test for configuring CRIU_LOG_LEVEL on checkpoint

Up to now the CRIU_LOG_LEVEL env variable was only
recognized for the restore side. This change enabled it
so that the criu log level can be set on the checkpoint side.

njr-11 committed:
Merge pull request #22309 from njr-11/22308-PersistentExecutorTimersWithFailoverEnabledTest-infrastructure-slowness-workaround

PersistentExecutorTimersWithFailoverEnabledTest infrastructure workaround

njr-11 committed:
Merge pull request #22313 from njr-11/22312-testRetryFourWithOneSkipWithFailoverEnabled-times-out-after-2-seconds

testRetryFourWithOneSkipWithFailoverEnabled times out after 2 seconds

jhanders34 committed:
Create MP JWT 2.1 feature files based off of PR #22083

- Create mpJwt 2.1 public and API features
- Have mpJwt 2.1 not pull in appSecurity feature, but instead update
servlet, restfulWS and CDI auto feature to include mpJwt 2.1

pmd1nh committed:
FAT for ServletRequest Id and ServletConnection

jim-krueger committed:
Changes needed to allow restfulWS30 tests to repeat for EE10

dazey3 committed:
Add new JPA 3.1 tests for Expressions as conditions in Criteria CASE expressions

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

acdemyers committed:
Call secureRandom in LTPACrypto
extra cleanup

jhanders34 committed:
Enable mpJwt 2.1 tests

Azquelt committed:
Merge pull request #22290 from Azquelt/microprofile6-mpMetrics-noee9

Make mpMetrics-5.0 dummy feature not tolerate EE9

volosied committed:
Skip testNegativeoTimeOut

sebratton committed:
Remove unnecessary repeater rules

Remove EE repeater where the use of the EE app is incidental to what the FAT is testing and where the same app is already repeated in other FAT tests

tevans78 committed:
Merge pull request #22294 from tevans78/mp60Update

Update MP6 feature

jgrassel committed:
Merge pull request #22314 from jgrassel/UpdateJPA31FAT_FourthTime

Update JPA 3.1 FAT, fix incorrect column types on DB2 and Oracle DDL …

Azquelt committed:
Merge pull request #22322 from Azquelt/mpOpenAPI-3.1-configValidation

Update mpOpenAPI 3.1 for config validation

tjwatson committed:
Merge pull request #22333 from tjwatson/checkpointCRIULogLevel

Test setting CRIU_LOG_LEVEL for checkpoint

Commit:
Add Decryption support to ExternalTestService.getProperty()

tevans78 committed:
Merge pull request #22345 from websterc87/ExternalTestService

Add Decryption support to ExternalTestService.getProperty()

emilytee committed:
JakartaEE10Authorization-build.gradle

pnicolucci committed:
MYFACES-4449 updates

pnicolucci committed:
Add oam.lifecycle.clientwindow to thread context

pnicolucci committed:
Disable call to getAttribute until implementated

pnicolucci committed:
Update SimpleTest to use Faces 4.0

ayoho committed:
Issue 21165: Add additional support for verifying OIDC callback

Adds the following validation to the callback from the OP:
- Validates the callback URI to ensure it matches the configured redirect URI or original request URI
- Validates the state value
- Validates the callback request does not include an `error` parameter

For #21165

c00crane committed:
Wait for mongo to be ready before registering clients

Azquelt committed:
Run MP FT 4.0 TCK with just core EE features

ayoho committed:
Remove old GSON reference from dependabot pom.xml

neuwerk committed:
Merge branch 'OpenLiberty:integration' into removeCXF262Bundles

lavenal committed:
Fix locale selection

Commit:
Update OidcHttpAuthenticationMechanism to start processing callback for issue #21165

c00crane committed:
Merge remote-tracking branch 'origin/integration' into jakartaFat-1

neilgsyoung committed:
complete db transactions

pmd1nh committed:
Merge pull request #22319 from pmd1nh/20417-Cookie_SetAttribute_FAT

FAT for Cookie setAttribute

sebratton committed:
Merge pull request #22344 from sebratton/remove-repeater-rule

Remove unnecessary repeater rules

njr-11 committed:
Merge pull request #22332 from njr-11/21063-annotation-updates-for-records

updates to entity annotations for records

teddyjtorres committed:
Merge pull request #22359 from teddyjtorres/21165-continueFlow

Update OidcHttpAuthenticationMechanism to start processing callback for issue #21165

jimmy1wu committed:
init userinfo subsystem (#22275)

* init userinfo subsystem

* address feedback

* add message to assertNull

c00crane committed:
Merge pull request #22353 from c00crane/291991.1

Wait for mongo to be ready before registering clients

volosied committed:
Merge pull request #22343 from volosied/fix-wsoc-timeout

Skip testNegativeoTimeOut in wsoc fat

tjwatson committed:
Fix ConcurrentModificationException for notifyListeners

Some listeners would remove themselves when being called.
This would lead to exceptions when iterating over the
listeners.

The fix is to make a copy of the listener list first and
not iterate over the live list.

pnicolucci committed:
Merge pull request #22350 from pnicolucci/MYFACES-4449-Plus-OtherUpdates

Myfaces 4449 plus other updates

BerksanAtes committed:
Duplicate copy of io.openliberty.mail.2.0.internal_fat with mail.2.1 bundles

jhanders34 committed:
Merge pull request #22338 from jhanders34/perf-improvement3

Create MP JWT 2.1 feature files based off of PR #22083

neuwerk committed:
Merge pull request #22005 from neuwerk/removeCXF262Bundles

Remove CXF 2.6.2 and related bundles

anjumfatima90 committed:
Add tests for appSecurity with checkpoint

jhanders34 committed:
Merge pull request #22193 from jhanders34/perf-improvement6

Update transformer to version 0.5.0

sebratton committed:
Merge pull request #22335 from sebratton/fix-eclipse-compile-err

Add to bnd buildpath to address an eclipse compile issue.

Commit:
Issue #19456: Configure infinispan to run with and without near-side cache randomly each run. Update FAT tests to not run if skip.test system property is set. This will help with the eXtreme Scale fat in WS-CD-Open which can't run with Java 11 or on zOS at this time.

c00crane committed:
Merge remote-tracking branch 'origin/integration' into jakartaFat-1

pnicolucci committed:
MYFACES-4451 initial checkin

pnicolucci committed:
MYFACES-4451: remove deprecated constructor

pnicolucci committed:
MYFACES-4452: initial checkin

pnicolucci committed:
MYFACES-4452: getSubIdPattern should be private

dazey3 committed:
Add JPA 3.1 tests for EclipseLink Query FAT

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

jhanders34 committed:
Remove projects that are no longer needed

- Remove projects used with CXF 2.6.2 for wssecurity
- Remove 2.3 version of the JAXB and JAXWS API
- Remove 3.2 version of wsat.cxf.utils.3.2 project

jhanders34 committed:
Update projects and features to not reference removed projects.

- Update projects' buildpath to remove references to removed projects
- Update use of jaxws23 to be jaxws from rename of project.
- Update to use 3.1.1 version of neethi with cxf 3.2 for jaxwsClient
- Remove features that aren't used any more.
- Update features to remove reference to wsat.cxf.utils.3.2
- Update code to not use wsat.cxf.utils.3.2, but instead to call the cxf
code directly now that there is only one cxf version.
- Update saml.sso tests to use JDK Base64 instead of xmltooling

jhanders34 committed:
Remove removed artifacts from dependency maven files.

Commit:
Merge pull request #22331 from jonhawkes/test-ordering

Ignore invalid resources in checkAtomicity()

jim-krueger committed:
Merge pull request #22339 from jim-krueger/RESTful-FAT-EE10

Changes needed to allow restfulWS30 tests to repeat for EE10

tjwatson committed:
Merge pull request #22365 from tjwatson/fixCpuInfoCheckpoint

Fix ConcurrentModificationException for notifyListeners

pnicolucci committed:
Merge pull request #22373 from pnicolucci/MYFACES-4451

Myfaces 4451

pmd1nh committed:
Merge pull request #22337 from pmd1nh/20425-FAT_Request_ConnectionID

FAT for ServletRequest Id and ServletConnection

joe-chacko committed:
Merge pull request #22234 from joe-chacko/22227-yoko-null-fields

Issue #22227: update Yoko for null field interoperability

tkburroughs committed:
Reduce EJB FAT repeats on Windows
- reduce the number of repeat actions on windows to avoid timeout

joe-chacko committed:
Merge pull request #21864 from joe-chacko/messaging-singletons

Improve messaging singleton mechanism

anjumfatima90 committed:
Repeat the tests for jakartaee9.

tjwatson committed:
Fix message for invalid checkpoint phase

When an invalid checkpoint phase is specified the server
returns very fast.  This maybe too fast for tail to respond
to the output message.  This just makes sure we wait 2 seconds
if the return code is 20. 20 is used for invalid arguments.

neuwerk committed:
Put mail-2.1 into beta

dazey3 committed:
Enable existing JPA FAT testing for JPA 3.1

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

BerksanAtes committed:
WIP

tkburroughs committed:
Merge pull request #22381 from tkburroughs/ReduceEJBFATRepeatsOnWindows

Reduce EJB FAT repeats on Windows

dazey3 committed:
fix minor Oracle failure

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

jhanders34 committed:
Merge pull request #22378 from jhanders34/perf-improvement3

Remove additional projects no longer used with CXF 2.6.2 removal

jhanders34 committed:
Merge pull request #22382 from neuwerk/mail21Beata

Jakarta Mail 2.1: Change mail-2.1 feature to kind=beta

pnicolucci committed:
Initial JSF 2.3 FAT EE10 enablement

pnicolucci committed:
MYFACES-4454: initial checkin

pnicolucci committed:
MYFACES-4454: updates

pnicolucci committed:
MYFACES-4453: initial API checkin

pnicolucci committed:
MYFACES-4453: API updates

pnicolucci committed:
MYFACES-4453: initial IMPL checkin

pnicolucci committed:
MYFACES-4453: IMPL updates

neilgsyoung committed:
Merge pull request #22262 from neilgsyoung/27592-splitlonghadbsuite

Split out HADB retriable sqlcode tests

neilgsyoung committed:
Merge pull request #22167 from neilgsyoung/21816-leasedeleteonshutdown

Dont check server stopping early in lease deletion

Azquelt committed:
Merge pull request #22354 from Azquelt/microprofile6-mpFT-tck-noservlet

Run MP FT 4.0 TCK with just core EE features

pnickoll committed:
Adding NLS returns for 22.0.0.10

Azquelt committed:
Update CDI Lifecycle FAT and repeat on EE10

Azquelt committed:
Add a public method to check a FATServlet response

This allows the parsing, checking and deserialization of an
AssertionError for a response from a FATServlet to be called from a test
class.

I needed to use WebBrowser to call several tests within the same HTTP
session.

c00crane committed:
Merge pull request #22326 from c00crane/jakartaFat-1

Jakarta security fat setup

jim-krueger committed:
Merge pull request #22181 from jim-krueger/Remove-Woodstox-warning

Remove warning since Liberty does not ship Woodstox

ayoho committed:
Merge pull request #22336 from ayoho/jakarta-21157-discoverySsl

PR for Issue 21157: Add SSL support for discovery

ayoho committed:
Merge pull request #22329 from ayoho/jakarta-22300

PR for Issue 22300: Fix issue with empty strings returned from provider metadata

ayoho committed:
Merge pull request #22355 from ayoho/removeGson

Remove old GSON reference from dependabot pom.xml

lavenal committed:
fix ToolboxAPIv1Test.catalogRemoveCausesToolboxRemove if running as the first test

njr-11 committed:
Issue #18228 Asynchronous on REST application method

Commit:
Register OidcIdentityStoreBean for issue #21169

Azquelt committed:
Update CDI non-contextual FAT and run on EE10

dazavala committed:
Fix spiVisibility attribute name and desc

pmd1nh committed:
Beta Servlet 6.0

ayoho committed:
Issue 21165: Remove dependency on Java EE security; NLS updates

jhanders34 committed:
Clean up remaining JAXB and JAXWS 2.3 references

- Remove com.ibm.ws.jaxws.2.3.wsat
- Update features to not tolerate non existent 2.3 jaxb feature
- Remove jaxwsClient 2.3 private feature.

njr-11 committed:
Issue #21063 wildcards in Like keyword

arunavemulapalli committed:
oidc identitystore token validation (2)

update oidc server, so it can use common utilities

pnicolucci committed:
Merge pull request #22375 from pnicolucci/MYFACES-4452

Myfaces 4452

pnicolucci committed:
Merge pull request #22386 from pnicolucci/MYFACES-4454

Myfaces 4454

cbridgha committed:
22.0.0.10 Notices

jhanders34 committed:
Merge pull request #22403 from jhanders34/perf-improvement3

Clean up remaining JAXB and JAXWS 2.3 references

teddyjtorres committed:
Merge pull request #22394 from teddyjtorres/21169-cdiInstallsIdentityStoreBean

Register OidcIdentityStoreBean for issue #21169

Azquelt committed:
Merge pull request #22391 from Azquelt/cdi-ee10-lifecycle-fat

Update CDI Lifecycle FAT and repeat on EE10

emilytee committed:
CodeReviewUpdates

cbridgha committed:
Merge pull request #22407 from cbridgha/Notices_220010

22.0.0.10 Notices

ayoho committed:
Merge remote-tracking branch 'origin/integration' into jakarta-21165-1

neuwerk committed:
Add the SOAP 3.0 implementation

emilytee committed:
CodeReviewUpdates2

olendvcook committed:
EJB Persistent timer test - ignore shut down error

KyleAure committed:
Ensure FFDCs are not created when parsing deployment descriptor (#22395)

Azquelt committed:
Update CDI thirdparty FAT and run on EE10

dazavala committed:
Merge pull request #22398 from dazavala/fixBellOsgiInfProps

Fix spiVisibility attribute name and desc post-translation

anjumfatima90 committed:
Merge pull request #22376 from anjumfatima90/checkpoint-appSecurity

Add tests for appSecurity with checkpoint

tjwatson committed:
Merge pull request #22401 from tjwatson/allowInvalidPhaseMessage

Fix message for invalid checkpoint phase when in container

njr-11 committed:
Issue #21063 contains for non-collection types

pnicolucci committed:
Merge pull request #22387 from pnicolucci/MYFACES-4453

Myfaces 4453

jjiwooLim committed:
Merge pull request #22327 from jjiwooLim/featureUtility-connectionError

featureUtility repo connection failure error code

BerksanAtes committed:
Jakarta Mail 2.1 tests done

lavenal committed:
Merge pull request #22379 from lavenal/fix-locale-selection-adminCenter

AdminCenter and its tools: change locale selection based on the browser language settings

lavenal committed:
Merge pull request #22392 from lavenal/fix-toolboxAPITest-execution-order-problem

Fix ToolboxAPIv1Test.catalogRemoveCausesToolboxRemove test

pmd1nh committed:
Merge pull request #22400 from pmd1nh/21528-Beta_Servlet60

Beta Servlet 6.0

jdmcclur committed:
change cache method and reorder building of route template

BerksanAtes committed:
Copyright changes

loriadi committed:
PH46816: Header validation

ayoho committed:
Issue 21165: Add unit tests for OIDC response validation

loriadi committed:
PH46816: copyright

ayoho committed:
Issue 21157: Perform discovery for every OIDC request, not just for missing authz endpoint

Updates the Jakarta Security code to perform OIDC Discovery right at the start of each flow. Before, we were only performing discovery when the authorization endpoint URL wasn't configured in the provider metadata in the servlet annotation. This will ensure we always have the OP's metadata and that values in the provider metadata can be used to override values from the discovery data.

For #21157

ayoho committed:
Update io.openliberty.org.testcontainers/cache/projects

BerksanAtes committed:
Resolved review changes

Azquelt committed:
Merge pull request #22406 from Azquelt/cdi-ee10-noncontextual-fat

Update CDI non-contextual FAT and run on EE10

Azquelt committed:
Merge pull request #22414 from Azquelt/cdi-ee10-thirdparty-fat

Update CDI thirdparty FAT and run on EE10

joe-chacko committed:
Use varargs in SibTr and SibMessage

loriadi committed:
PH46816 Update

emilytee committed:
CodeReviewChgs3

BerksanAtes committed:
Minimum Java level is 11

pnicolucci committed:
MYFACES-4457 UIWebSocket initial checkin

pnicolucci committed:
MYFACES-4457: UIWebSocket getEventNames signature fix

kristip17 committed:
Issue 21156: Add EL to additional wrapper class for jakarta security 30

ayoho committed:
Issue 21166: Send token request to OP

For #21166

BerksanAtes committed:
Gradle change

jdmcclur committed:
changes based on Jared's review

BerksanAtes committed:
PB issue resolved

tevans78 committed:
Merge pull request #22388 from pnickoll/PII_22.0.0.10

Adding NLS returns for 22.0.0.10

tevans78 committed:
Merge pull request #22426 from loriadi/PH46816-validation

PH46816: Header validation

tevans78 committed:
remove token from logs

emilytee committed:
JakartaEE10Authorization-copyrights

ayoho committed:
Issue 21166: Enhance discovery code for authz and token requests

For #21166

BerksanAtes committed:
Removal of third party class from test

jhanders34 committed:
Apply suggestions from code review

Commit:
Issue #19609: Upgrade Hazelcast version for JCache tests to 5.1.3 from 3.12.12.

jjiwooLim committed:
round progress bar status

tevans78 committed:
Merge pull request #22448 from tevans78/noTokenLog

remove token from logs

acdemyers committed:
Merge pull request #22340 from acdemyers/292319-ltpa

Call secureRandom in LTPACrypto

tjwatson committed:
Fix immutable typo

emilytee committed:
Merge pull request #22158 from emilytee/JakartaEE10Authorization

Jakarta ee10 authorization

joe-chacko committed:
Merge pull request #22436 from joe-chacko/messaging-singletons

Use varargs in SibTr and SibMessage

tabishop committed:
ignore reg change timeout error msg (#22402)

tabishop committed:
remove tests duplicated in com.ibm.ws.jca_fat (#22419)

BerksanAtes committed:
New FAT bukect is added

Commit:
Merge pull request #22370 from jvanhill/19456-JCacheNearCacheTests

Issue #19456: Configure infinispan to run with and without near-side …

ayoho committed:
Merge pull request #22399 from ayoho/jakarta-21165-1

PR for Issue 21165: Add additional support for verifying OIDC callback

ayoho committed:
Merge pull request #22427 from ayoho/jakarta-fixDiscovery

PR for Issue 21157: Perform discovery for every OIDC request, not just for missing authz endpoint

olendvcook committed:
Merge pull request #22412 from olendvcook/EJBPersistentTimerIgnoreShutdownError

EJB Persistent timer test - ignore shut down error

ayoho committed:
Merge pull request #22428 from ayoho/updateTestcontainersProjects

Update io.openliberty.org.testcontainers/cache/projects

Commit:
Merge pull request #22451 from jvanhill/19609-Hazelcast5

Issue #19609: Upgrade Hazelcast version for JCache tests to 5.1.3 fro…

jimmy1wu committed:
fix base url used in OpenIdAuthenticationMechanismDefinition (#22430)

fmhwong committed:
Add write permission

tjwatson committed:
First wait for WAB deployment before acquiring static hooks

The WAB deployment triggers the job to do plugin generation
for the endpoints.  The plugin generation registers a static
hook that delays the plugin generation until the process is
restored.  Before it was possible that the WABInstaller
checkpoint hook would get called after acquiring the static
hooks.  Once this happens no new static hooks are allowed to
be added to the CheckpointPhase class.

The issue is that some background thread may be still deploying
the system WABs and have yet to trigger the plugin generator.
This means the plugin generator may not have registered its hook
yet.  Ultimately this leads to the plugin generator not being
able to register its hook so it resorts to immediate plugin
generation on the checkpoint side.  This fix ensures that
all system WABs have been deployed before calling any static
checkpoint hooks.

njr-11 committed:
Merge pull request #22393 from njr-11/18228-Asynchronous-on-REST-method

Asynchronous on REST application method

njr-11 committed:
Merge pull request #22417 from njr-11/21063-like-keyword

like keyword and contains for non-collections

gjwatts committed:
Merge pull request #22328 from gjwatts/java-19-FAT-test

Java 19 FAT test

gjwatts committed:
Open com.sun.jndi.dns to jdk.naming.dns module

Commit:
Fix for test failures in RTC 291613. No longer will emit FFDC for failures in deactivate methods in CacheServiceImpl, CacheManagerServiceImpl and CachingProvider.

arunavemulapalli committed:
Merge pull request #22408 from arunavemulapalli/jakarta-sec30-identitystore-1

Jakarta security 30 identitystore implementation

contbld committed:
Update product and bundle version for release 22.0.0.11

pnicolucci committed:
Merge pull request #22442 from pnicolucci/MYFACES-4457

MYFACES-4457 UIWebSocket getEventNames signature fix

pnicolucci committed:
Merge pull request #22385 from pnicolucci/InitialJSF23EE10Enablement

Initial JSF 2.3 FAT EE10 enablement

Commit:
Ensure test server doesn't leak

tspewak committed:
Merge pull request #22304 from tspewak/22303-AllowAttachToWorkOnZosWithJava11

Issue #22303: On z/OS running Java 11 a FFDC with caused by AttachNotSupportedException occurs when feature localConnector-1.0  is specified.

jdmcclur committed:
Merge pull request #22248 from jdmcclur/otel-perf

Cache routes to avoid looking up the uri info every request.

fmhwong committed:
Merge pull request #22458 from fmhwong/22456-fixpermission

Add write permission to javapermission

BerksanAtes committed:
Merge pull request #22418 from BerksanAtes/22114-Jakarta_Mail_2.1_Tests

Jakarta Mail 2.1 tests

neilgsyoung committed:
not finding lease on claim is allowable

Commit:
Merge pull request #22462 from jvanhill/rtc291613-2

Fix for test failures in RTC 291613. No longer will emit FFDC for fai…

ayoho committed:
Merge pull request #22454 from ayoho/jakarta-21165-2

PR for Issue 21165: Add unit tests for OIDC response validation

jimmy1wu committed:
implement session based storage (#22450)

* implement session based storage

* add note and test about storage properties

volosied committed:
Merge pull request #22296 from volosied/20584-upgradeHttpToWebSocket-20585-getRequestURI

WebSocket 2.1: Implement upgradeHttpToWebSocket and return full URI from getRequestURI

gjwatts committed:
Merge pull request #22461 from gjwatts/22361-Open-com.sun.jndi.dns

Open com.sun.jndi.dns to jdk.naming.dns module

jjiwooLim committed:
Merge pull request #22452 from jjiwooLim/featureUtility-progressbar

FeatureUtility round progress bar percentage

ayoho committed:
Revert "implement session based storage (#22450)"

This reverts commit 925ad1ddd3aa7a4052c4d7b68788b7fb3ce88d4b.

ayoho committed:
Merge pull request #22473 from OpenLiberty/revert-22450-21162_sessionBasedStorageImpl

Revert "implement session based storage"

ayoho committed:
Revert "Revert "implement session based storage""

ayoho committed:
Fix compilation issue in JakartaOidcAuthenticationResponseValidator

arunavemulapalli committed:
preparing to share jwk between oidc and jakarta30 security

explicitely set default ssl use with jwk retriever

tjwatson committed:
Merge pull request #22459 from tjwatson/fixWABCheckpointHook

Fix wab checkpoint hook

tjwatson committed:
Update GLIBC_TUNABLES to fix portability issues

The usable options to GLIBC_TUNABLES cause issues when
doing a checkpoint on AMD and restoring on Intel or
vise-versa. This updates the setting to remove the
_usable options.

tjwatson committed:
For checkpoint add the -Djdk.nativeEC=false option

There are portability issues with nativeEC when doing
a checkpoint on AMD and restoring on Intel.  The reverse
doesn't seem to be an issue.  disabling nativeEC in the
JDK works around the issue.  This should not impact
performance to restore.

This is also a temporary setting while we work out the
real fix.

kristip17 committed:
Merge pull request #22464 from kristip17/UpdateELForAuthMechWrappers

Issue 21156: Add EL to additional wrapper class for jakarta security 30

kristip17 committed:
Issue 22465: Add EL support for OpenIdProviderMetadataWrapper

dazey3 committed:
Merge pull request #22230 from dazey3/splitJPA20FAT_integration

Split `com.ibm.ws.jpa.tests.spec20_fat` FAT

Commit:
Merge pull request #22469 from jonhawkes/cleanup-longlease-server

Ensure test server doesn't leak

Yas-Franco committed:
Add mpTelemetry TCK (#22389)

Add MP Telemetry tck

neilgsyoung committed:
Merge pull request #22323 from neilgsyoung/22321-clearupstalelease

Tidy up stale lease in tx failover tests

tevans78 committed:
Merge pull request #22468 from OpenLiberty/autoVersionUpdate_20220913-0408

Update product and bundle version for release 22.0.0.11

ayoho committed:
Merge pull request #22475 from OpenLiberty/revert-22473-revert-22450-21162_sessionBasedStorageImpl

Revert "Revert "implement session based storage""

ayoho committed:
Merge pull request #22471 from ayoho/jakarta-21166-discoveryUpdate

PR for Issue 21166: Send token request to OP and enhance discovery

kristip17 committed:
Merge pull request #22482 from kristip17/addELSupportOpenIdProvMetadata

Issue 22465: Add EL support for OpenIdProviderMetadataWrapper

dazey3 committed:
Merge pull request #22384 from dazey3/enableJPA31Testing_integration

New JPA 3.1 testing

neuwerk committed:
Merge pull request #22410 from neuwerk/saaj30

Jakarta XML Web Services 4.0: Add the SOAP 3.0 implementation

arunavemulapalli committed:
Merge pull request #22479 from arunavemulapalli/jakarta-sec30-oidc-jwk

preparing to share jwk utilities between oidc and jakarta30 security and other minor updates

tjwatson committed:
Merge pull request #22481 from tjwatson/fixPortableCheckpoint

Fix portable issues with checkpoint and restore across different chipsets

Commit:
Merge branch 'opentracing' of github.com:dettmoney/open-liberty into opentracing
